src/collection-view.js

```diff
@@ -321,9 +337,19 @@ Marionette.CollectionView = Marionette.View.extend({
     // remove and/or close it later
     this.children.add(view);
 
+    if (canTriggerAttach && this.triggerBeforeAttach) {
+      nestedViews = [view].concat(view._getNestedViews());
+      // Trigger before:attach after render to honor Region#show() lifecycle order.
+      view.once('render', _.partial(this._triggerMethodOnAll, nestedViews, 'before:attach'), this);
```

Comment:
I see your point about the lifecycle event in a public API method, although I'm not _perfectly_ comfortable with it.  I guess it's a lesser of two evils, eh?  But I _am_ concerned that adding not only a lifecycle event, but a lifecycle event contingent on a private variable, could throw some people for a loop--especially since changing the private variable would mean legitimate overrides could break.